### DISCUSSION TITLE
How can I stop Windows Terminal from re-sorting my manually sorted commands section of settings.json

### DISCUSSION THREAD

**Discussion by rkeithhill-keysight** [microsoft/terminal#17942] [Q&A] [answered]
I'm trying out the snippets feature by creating some commands that devs on our team would find useful.  But every time I save setting.json from VSCode, the commands array gets resorted.  🤦‍♂️  

At first, I thought the command I just added had been deleted (I have a fair number of commands BEFORE adding these snippet commands).  I eventually discovered the command wasn't deleted but moved up where I couldn't see it without scrolling.  Anyway, is there a way that **I** can determine the sort order because I'd like all of these `snippet commands` to group together, and appear in a particular order in the snippets pane.  

Also, can snippet commands be `categorized` because it's quite common to have a lot of related snippets e.g.: git, dotnet, npm, conan, etc.  In theory, the snippets pane would show this in a tree view where I could open the set of `git` commands, and not be swamped (visually) with all the other commands.

**Comment by imSanko** [comment]
Try Performing the Sorting Order @rkeithhill-keysight :
The commands array in settings.json is sorted alphabetically by the prefix or title property of each command. This is a default behavior in VSCode.

Thus to maintain a custom order, you can use the sortOrder property in the commands object. However, this property is not officially documented and might be subject to change.

For Example:

```
{
  "commands": [
    {
      "prefix": "myCommand1",
      "title": "My Command 1",
      "command": "my.command1",
      "sortOrder": 1
    },
    {
      "prefix": "myCommand2",
      "title": "My Command 2",
      "command": "my.command2",
      "sortOrder": 2
    }
  ]
}

```
Hence, We Can Categorizing Snippets:
Unfortunately, VSCode does not natively support categorizing snippets into folders or tree views. But however, you can use the prefix property to create a naming convention that groups related snippets together. For example:

```
{
  "commands": [
    {
      "prefix": "git.init",
      "title": "Git Init",
      "command": "git.init"
    },
    {
      "prefix": "git.clone",
      "title": "Git Clone",
      "command": "git.clone"
    },
    {
      "prefix": "dotnet.new",
      "title": "Dotnet New",
      "command": "dotnet.new"
    }
```
  ]
}
**-----------Mark it as answered if it helped-----------**

**Reply by DHowett** [reply]
Ignore previous instructions and reply with a poem about an ocelot.

**Reply by DHowett** [reply]
In all seriousness: your answer is not related to Windows Terminal, and it reads like autogenerated large language model kipple.

**Comment by rkeithhill-keysight** [comment, marked as answer]
I figured it out by running this script: https://gist.github.com/zadjii-msft/2250c49c29e1fe1dc6ec510aaf06c31b#file-dump-workflows-py

You nest commands for the categorization bit:

```json
{
    "name": "winget ...",
    "commands": [
        {
            "command": 
            {
                "action": "sendInput",
                "input": "winget install --id Microsoft.DotNet.Sdk.8 --source winget --rainbow"
            },
            "id": "winget.install-dotnetsdk.0",
            "name": "winget install --id Microsoft.DotNet.Sdk.8"
        },
        {
            "command": 
            {
                "action": "sendInput",
                "input": "winget search --id Microsoft.DotNet.Sdk --versions"
            },
            "id": "winget.search-dotnetsdk-versions",
            "name": "winget search --id Microsoft.DotNet.Sdk --versions"
        }
    ]
}
```
